src/components/RadioGroup/types.ts

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { RadioGroupColor } from "../internal/InternalRadioGroup";
+
+type StringOption = {
+  label: string;
+  description?: React.ReactNode;
+  value: string;
+};
+
+type NodeOption = {
+  label: React.ReactNode;
+  description: undefined;
+  value: string;
+};
+
+export type RadioOption =
+ | StringOption
+ | NodeOption
+ | Readonly<StringOption>
+ | Readonly<NodeOption>;
+
+
+export type RadioOptions =
+ | StringOption[]
+ | NodeOption[]
+ | Readonly<StringOption>[]
+ | Readonly<NodeOption>[];
+
+export interface DeprecatedRadioGroupProps {
+  name: string;
+  label: string;
+  options: RadioOptions;
+  columns?: number;
+  color?: RadioGroupColor;
+  helpText?: string;
+  disabled?: boolean;
+  optional?: boolean;
+  validate?: boolean | ((value: any, data: any) => string | string[] | null);
+  validateData?: any;
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
+  testId?: string;
+}
+
+export interface InternalRadioGroupProps {
+  label: string;
+  options: RadioOptions;
+  columns?: number;
+  color?: RadioGroupColor;
+  helpText?: string;
+  disabled?: boolean;
+  optional?: boolean;
+  validate?: boolean | ((...args: any[]) => void);
```

Comment:
Shouldn't the validate type here be the same as in DeprecatedRadioGroupProps ?
And, why don't we use the ValidateFn type we have?

Reply:
Actually no it should not exist, as validate only exists on `Field` component it doesn't exist on the `RadioGroup`